### ISSUE TITLE
Bool type support in function signature encoding

### ISSUE THREAD

**Issue by humayunj** [elek/cethacea#1]
Thanks for this incredibly sharp swiss-army knife. I wonder if it's still being maintained. Straight to the issue.

The function arguments encoding doesn't support bool types. As we can see in  [EncodeArguments](https://github.com/elek/cethacea/blob/main/pkg/encoding/encoding.go#L18) function.

As the result ceth exits with `unsupported argument type { bool false}`. Case for `tuple` is not implemented either but I haven't dig into it.

**Example**

Command: `ceth contract call --debug --account=key1 "setApprovalForAll(address,bool)" $operator false // or 0/0x00`

Output: `Error: unsupported argument type { () false}`

**Comment by elek** [comment]
>  I wonder if it's still being maintained

Well, I use it, but usually I implement only what I directly need ;-) 

But bool support is easy. Just added it in 46f14da7ca680916b3921a2f24fa0109e561c6eb. 

Thanks for the report.

**Comment by humayunj** [comment]
That was quick! I actually found the tool from your Storj blogs, and used cethacea though whole process of deploying and debugging (tx debug) while implementing ERC-721 in Yul. Great work!

**Comment by elek** [comment]
Great. Thanks to share. Good to know.

We (storj) also uses it for some integration tests and development (for example in https://github.com/storj/storjscan). So it should definitely work in the future.

I just started to use `cethacea` for transferring real money (ETH/ tokens) from command line. That's the reason why I  added the confirmation option (and the gas calculation also should be fixed. Earlier I used a very high number which doesn't fly on real servers)...
